numpy/fft/helper.py

```diff
@@ -159,6 +160,8 @@ def fftfreq(n, d=1.0):
     """
     if not isinstance(n, integer_types):
         raise ValueError("n should be an integer")
+    if not isinstance(d, numbers.Number):
```

Comment:
```suggestion
    if np.ndim(d) != 0:
```
Is probably best, as that won't break `d=np.array(1)`

Reply:
Yes, I certainly agree. I have implemented the changes as per your suggestion. Thanks!